src/main/java/com/tabnine/binary/requests/autocomplete/CompletionPostprocess.kt

```diff
@@ -0,0 +1,48 @@
+package com.tabnine.binary.requests.autocomplete
+
+fun postprocess(request: AutocompleteRequest, result: AutocompleteResponse, tabSize: Int) {
+    result.results.forEach { it.new_prefix = it.new_prefix.replace("\t", " ".repeat(tabSize)) }
+
+    val requestIndentation = lastLineIndentation(request.before.replace("\t", " ".repeat(tabSize))) ?: return
+    if (requestIndentation == 0) return
+
+    val regex = constructRegex(requestIndentation)
+    result.results.forEach { entry ->
+        val calculateTrimmingIndex = calculateTrimmingIndex(entry.new_prefix, regex)
+        calculateTrimmingIndex?.let { entry.new_prefix = entry.new_prefix.take(it) }
+    }
+}
+
+/**
+ * Finds the amount of spaces or tabs in the last line of the given `value`,
+ * returning `null` if `value` has no newlines, or the last line is not whitespaces only.
+ */
+fun lastLineIndentation(value: String): Int? {
+    try {
+        val lastLine = value.lines().last()
+        if (lastLine.isBlank()) {
+            return lastLine.length
+        }
+    } catch (e: java.util.NoSuchElementException) {
```

Comment:
When is this exception raised? Upon no lines at all?

Reply:
![image](https://user-images.githubusercontent.com/67855609/166506259-274ac40d-7c83-4d1e-936f-478835e24a38.png)

Reply:
this is so stupid to throw an exception if the list is empty... why doesn't kotlin return `T?` 😢